internal/core/data/v2/application/event.go

```diff
@@ -19,25 +19,42 @@ import (
 	"github.com/edgexfoundry/go-mod-core-contracts/clients"
 	"github.com/edgexfoundry/go-mod-core-contracts/errors"
 	"github.com/edgexfoundry/go-mod-core-contracts/v2/dtos"
+	dto "github.com/edgexfoundry/go-mod-core-contracts/v2/dtos/requests"
 	"github.com/edgexfoundry/go-mod-core-contracts/v2/models"
 
 	msgTypes "github.com/edgexfoundry/go-mod-messaging/pkg/types"
 
 	"github.com/google/uuid"
 )
 
+func validateEvent(e models.Event, profileName string, deviceName string, ctx context.Context, dic *di.Container) errors.EdgeX {
+	if profileName == "" {
+		return errors.NewCommonEdgeX(errors.KindContractInvalid, "empty profile name not allowed", nil)
+	}
+	if deviceName == "" {
+		return errors.NewCommonEdgeX(errors.KindContractInvalid, "empty device name not allowed", nil)
+	}
+	if e.ProfileName != profileName {
+		return errors.NewCommonEdgeX(errors.KindContractInvalid, fmt.Sprintf("event's profileName %s mismatches %s", e.ProfileName, profileName), nil)
+	}
+	if e.DeviceName != deviceName {
+		return errors.NewCommonEdgeX(errors.KindContractInvalid, fmt.Sprintf("event's deviceName %s mismatches %s", e.DeviceName, deviceName), nil)
+	}
+	return checkDevice(e.DeviceName, ctx, dic)
+}
+
 // The AddEvent function accepts the new event model from the controller functions
 // and invokes addEvent function in the infrastructure layer
-func AddEvent(e models.Event, ctx context.Context, dic *di.Container) (id string, err errors.EdgeX) {
-	configuration := dataContainer.ConfigurationFrom(dic.Get)
-	dbClient := v2DataContainer.DBClientFrom(dic.Get)
-	lc := container.LoggingClientFrom(dic.Get)
-
-	err = checkDevice(e.DeviceName, ctx, dic)
+func AddEvent(e models.Event, profileName string, deviceName string, ctx context.Context, dic *di.Container) (id string, err errors.EdgeX) {
```

Comment:
Why do we return id, since it is now always in the Event already?

Reply:
Right, no need to return the event id from application layer.  Removed the id.

Reply:
You're right. We don't have to return the id from the application layer.